File: benchmarks/fixtures/polyglot-services/src/user_service.lua
local function display(name, email)
    return string.format("%s <%s>", name, email)
end

local function create(params)
    return { name = params.name, email = params.email }
end

local user = create({ name = "Ivan", email = "ivan@example.com" })
print(display(user.name, user.email))

File: benchmarks/fixtures/polyglot-services/src/user_service.ex
defmodule UserService do
    defstruct name: "Alice", email: "alice@example.com"
    
    def display(user) do
        "#{user.name} <#{user.email}>"
    end
    
    def create(params) do
        struct(__MODULE__, params)
    end
end

user = %UserService{name: "Bob", email: "bob@example.com"}
IO.puts(UserService.display(user))